lib/aws/client.ex

```diff
@@ -12,9 +12,8 @@ defmodule AWS.Client do
             proto: "https",
             port: "443",
             http_client: {HTTPoison, []},
-            json_module: {Poison, [encode: :encode!, decode: :decode!]},
-            xml_module: {AWS.Util, [encode: :encode_xml, decode: :decode_xml]}
-
+            json_module: {Poison, []},
```

Comment:
This should look like this now:

```suggestion
            json_module: {AWS.JSON, []},
```

And the default implementation calls Poison. The same for XML and the HTTP client. They will call the behaviour which will also set a default implementation. For most cases though, the default implementation is simply a function call, so that's easy. :)